packages/perspective-jupyterlab/test/jupyter/widget.spec.js

```diff
@@ -105,37 +119,351 @@ utils.with_jupyterlab(process.env.__JUPYTERLAB_PORT__, () => {
+                            columns: {},
+                            editable: false,
+                            scroll_lock: true,
+                        },
+                        settings: true,
+                        sort: [],
+                        split_by: [],
+                        theme: "Material Light",
+                    });
+
+                    // Now restore in the viewer
+                    await viewer.evaluate(async (viewer) => {
+                        viewer.restore({
+                            columns: ["ui8"],
+                            filter: [["i8", "<", "50"]],
+                            group_by: ["date"],
+                            plugin: "X Bar",
+                            settings: false,
+                            sort: [["date", "asc"]],
+                            split_by: ["bool"],
+                            theme: "Material Dark",
+                        });
+
+                        return "";
+                    });
+
+                    // assert in python or else
+                    const error_cells_dont_exist =
+                        await assert_no_error_in_cell(
+                            page,
+                            `
+assert w.plugin == "X Bar"
+assert w.columns == ["ui8"]
+assert w.filter == [["i8", "<", "50"]]
+assert w.group_by == ["date"]
+assert w.split_by == ["bool"]
+assert w.plugin_config == {}
+assert w.settings == False
+assert w.sort == [["date", "asc"]]
+assert w.theme == "Material Dark"`
+                        );
+                    expect(error_cells_dont_exist).toBe(true);
+                },
+                {timeout: 120000}
+            );
+
+            test.jupyterlab(
+                "Edit from frontend - end to end",
+                [
+                    'w = perspective.PerspectiveWidget({"a": [True, False, True], "b": ["abc", "def", "ghi"]}, index="b", plugin_config={"editable": True})',
+                    "w",
+                ],
+                async (page) => {
+                    const viewer = await default_body(page);
+
+                    // assert in python or else
+                    let error_cells_dont_exist = await assert_no_error_in_cell(
+                        page,
+                        `assert w.table.view().to_df().to_dict() == {'a': {0: True, 1: False, 2: True}, 'b': {0: 'abc', 1: 'def', 2: 'ghi'}}`
+                    );
+                    expect(error_cells_dont_exist).toBe(true);
+
+                    // Toggle some values in the frontend
+                    const bools = await page.$$(".psp-bool-type");
+
+                    // do synchronous
+                    for (let bool of bools) {
+                        bool.click();
+                        // wait a short time
+                        await new Promise((res) => setTimeout(res, 250));
+                    }
+
+                    // wait a small time
+                    await new Promise((res) => setTimeout(res, 1000));
```

Comment:
Same - these are silly also since you can listen for updates on the viewer or view

Reply:
How long do you wait for the changes to update from python to JS? Isn't this just another avenue for arbitrarily long wait times? If tell puppeteer to wait for the viewer to mutate, how long do you wait for it to timeout? If you make it too soon you'll introduce extra flakiness.

Reply:
Like I suggested, you can listen for updates on the `<perspective-viewer>` or `View` for one option.  A much more explicit option would be to await the presence of the `<table>` -> `<td>`  DOM structure for the editable cell with the appropriat checkbox visible?